### PR TITLE
Add jshint rules

### PR DESCRIPTION
**Pull request by pouretrebelle**
I have no strong feelings about this

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "globalstrict"  : false,
+  "camelcase"     : true,
+  "curly"         : true,
+  "eqeqeq"        : true,
+  "forin"         : false,
+  "freeze"        : true,
+  "immed"         : true,
+  "indent"        : 4,
```
Comment (lpil) [inline .jshintrc]: 4?

File: README.md [hunk 2]
```diff
@@ -45,5 +45,5 @@ is a good place to start.
 
 | Linter   | When to use       | Config         |
 | -------- | -----------       | -------------- |
-| jshint   | Existing projects |                |
+| jshint   | Existing projects | `.jsnhintrc`   |
```
Comment (lpil) [inline README.md]: Typo here

File: .jshintrc [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  "latedef"       : "nofunc",
+  "newcap"        : true,
+  "noarg"         : true,
+  "noempty"       : true,
+  "nonbsp"        : true,
+  "nonew"         : false,
+  "plusplus"      : false,
+  "quotmark"      : "single",
```
Comment (lpil) [inline .jshintrc]: I would prefer double

File: .jshintrc [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  "undef"         : true,
+  "unused"        : true,
+  "maxparams"     : 4,
+  "maxdepth"      : 3,
```
Comment (lpil) [inline .jshintrc]: Seems high

File: .jshintrc [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  "maxcomplexity" : 10,
```
Comment (lpil) [inline .jshintrc]: Seems high

File: .jshintrc [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  "maxlen"        : 100,
```
Comment (lpil) [inline .jshintrc]: Is this function length or line length? I prefer 80

File: .jshintrc (hunk 3 above)
Comment (pouretrebelle) [inline .jshintrc]: no you don't get a say in this

File: .jshintrc (hunk 6 above)
Comment (pouretrebelle) [inline .jshintrc]: sure 80 it is

File: README.md (hunk 2 above)
Comment (pouretrebelle) [inline README.md]: k this was dumb

File: .jshintrc (hunk 1 above)
Comment (pouretrebelle) [inline .jshintrc]: this is a deprecated setting, I'll get rid of it. Might be worth adding [jscs](https://github.com/jscs-dev/node-jscs) too?

**Comment by lpil** [comment]
Looks good

**Pull request merged**